### PR TITLE
Adding game view

### PR DESCRIPTION
**Pull request by guilhermeluisdomingues**

### CODE REVIEW

**Review by pietro1704 — APPROVED** [review]

File: challengeFinal/Coordinator/Coordinator.swift [hunk 1]
```diff
@@ -8,9 +8,10 @@
 import Foundation
 import UIKit
 
-protocol Coordinator {
+protocol Coordinator: AnyObject {
```
Comment (guilhermeShima) [inline challengeFinal/Coordinator/Coordinator.swift]: Why are you using AnyObject?

File: challengeFinal/Coordinator/Storyboarded.swift [hunk 2]
```diff
@@ -9,13 +9,13 @@ import Foundation
...
 
-extension Storyboarded where Self: UIViewController{
-    static func instantiate() -> Self{
+extension Storyboarded where Self: UIViewController {
+    static func instantiate(storyBoardName: String) -> Self {
         let id = String(describing: self)
-        let storyboard = UIStoryboard(name: "Main", bundle: Bundle.main)
-        return storyboard.instantiateViewController(withIdentifier: id) as! Self
+        let storyboard = UIStoryboard(name: storyBoardName, bundle: Bundle.main)
+        return storyboard.instantiateViewController(identifier: id) as! Self
```
Comment (guilhermeShima) [inline challengeFinal/Coordinator/Storyboarded.swift:19]: Force CAST GRRR

File: challengeFinal/Flow/Menu/Coordinator/MenuCoordinator.swift [hunk 3]
```diff
@@ -8,16 +8,31 @@
...
-    
+
     func start() {
-        let vc = ViewController.instantiate()
+        let vc = MenuViewController.instantiate(storyBoardName: "Menu")
         navigationController.pushViewController(vc, animated: false)
     }
+
+    func startGame() {
+//        let vc =
```
Comment (guilhermeShima) [inline challengeFinal/Flow/Menu/Coordinator/MenuCoordinator.swift]: Unused function

File: challengeFinal/Flow/Story/Coordinator/StoryCoordinator.swift [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+        self.navigationController = navigationController
+    }
+    
+    func start() {
+        let vc = StoryViewController.instantiate(storyBoardName: "Story")
+        print("started")
+        navigationController.pushViewController(vc, animated: false)
+    }
+    
+    func startGame() {
```
Comment (guilhermeShima) [inline challengeFinal/Flow/Story/Coordinator/StoryCoordinator.swift]: Unused function

File: challengeFinal/Coordinator/Storyboarded.swift (hunk 2 above)
Comment (guilhermeShima) [reply challengeFinal/Coordinator/Storyboarded.swift:19]: What you think about replace it with a guard and an error log?

File: challengeFinal/Flow/Story/View/StoryView.swift [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+        addSubview(stack)
+        return stack
+    }()
+
+    public init(with viewModel: StoryViewModel) {
+        self.viewModel = viewModel
+        super.init(frame: .zero)
+        backgroundColor = UIColor(red: 0.89, green: 0.88, blue: 0.85, alpha: 1.00)
+
+        imageView.topAnchor.constraint(equalTo: topAnchor).isActive = true
```
Comment (guilhermeShima) [inline challengeFinal/Flow/Story/View/StoryView.swift:47]: What you think about creating a function to setup view constraints?

File: challengeFinalComponents/ImageView/ImageView.swift [hunk 6]
```diff
@@ -15,9 +15,11 @@ public class ImageView: UIView {
     private lazy var imageView: UIImageView = {
         let iView = UIImageView()
         addSubview(iView)
-        iView.widthAnchor.constraint(equalToConstant: imageWidth).isActive = true
-        iView.heightAnchor.constraint(equalToConstant: imageHeight).isActive = true
+//        iView.widthAnchor.constraint(equalToConstant: imageWidth).isActive = true
```
Comment (guilhermeShima) [inline challengeFinalComponents/ImageView/ImageView.swift]: Remove comments?

File: challengeFinalTests/Components/RegularTextViewTests.swift [hunk 7]
```diff
@@ -0,0 +1,29 @@
+//
+//  RegularTextViewTests.swift
+//  challengeFinalTests
+//
+//  Created by Guilherme Domingues on 14/10/20.
+//
+
+import XCTest
```
Comment (guilhermeShima) [inline challengeFinalTests/Components/RegularTextViewTests.swift:8]: Nice 👍

File: challengeFinal/Coordinator/Coordinator.swift (hunk 1 above)
Comment (guilhermeluisdomingues) [reply challengeFinal/Coordinator/Coordinator.swift]: The Coordinator must be an Class. Changed to class object.

File: challengeFinal/Flow/Menu/Coordinator/MenuCoordinator.swift (hunk 3 above)
Comment (guilhermeluisdomingues) [reply challengeFinal/Flow/Menu/Coordinator/MenuCoordinator.swift]: done

File: challengeFinal/Flow/Story/Coordinator/StoryCoordinator.swift (hunk 4 above)
Comment (guilhermeluisdomingues) [reply challengeFinal/Flow/Story/Coordinator/StoryCoordinator.swift]: done

**Pull request merged**